reasoner_transpiler/matching.py

```diff
@@ -166,16 +166,13 @@ def __init__(
         self.predicates: List[str] = ensure_list(
             edge.pop("predicates", []) or []
         )
-        # "related_to" is equivalent to no predicate
-        if self.predicates == ["biolink:related_to"]:
+
+        # "related_to" has every predicate as a descendent, it's inclusion is equivalent to no/any predicate
+        if 'biolink:related_to' in self.predicates:
             self.predicates = []
-        if primary_ks_required and not edge.get('_length'):
-            self.filters = [
-                f"`{edge_id}`.`biolink:primary_knowledge_source` IS NOT NULL"
-                ]
-        else:
-            self.filters = []
-        self.qualifier_filters = []
+
+        self.filters = []
+        self.qualifier_filters = ""
```

Comment:
Should this be [] ?

Reply:
It arguably should be in case we wanted to perform some operations on all of the filters in the future, but right now qualifier_filters and other filters are implemented differently - qualifier_filters is a string and filters is a list. I made this change to reflect that and to prevent type mismatch warnings in IDEs.